src/front-door/azext_front_door/_params.py

```diff
@@ -263,3 +266,59 @@ def load_arguments(self, _):
+        c.argument('header_action', arg_group="Action", arg_type=get_enum_type(HeaderActionType), help='Header action type for the requests.')
+        c.argument('header_name', arg_group="Action", help='Name of the header to modify.')
+        c.argument('header_value', arg_group="Action", help='Value of the header.')
+        c.argument('match_variable', arg_group="Match Condition", arg_type=get_enum_type(RulesEngineMatchVariable), help='Name of the match condition.')
+        c.argument('operator', arg_group="Match Condition", arg_type=get_enum_type(RulesEngineOperator), help='Operator of the match condition.')
+        c.argument('match_values', arg_group="Match Condition", nargs='+', help='Space-separated list of values to match against.')
+        c.argument('selector', arg_group="Match Condition", help='Optional selector for the match condition variable.')
+        c.argument('negate_condition', arg_group="Match Condition", arg_type=get_three_state_flag(), help='Applies "Not" to the operator.')
+        c.argument('transforms', arg_group="Match Condition", nargs='+', arg_type=get_enum_type(Transform), help='Space-separated list of transforms to apply.')
+        c.argument('priority', help='The priority number must start from 0 and consecutive. Rule with greater priority value will be applied later.')
+        c.argument('match_processing_behavior', arg_type=get_enum_type(MatchProcessingBehavior), help='Whether to stop processing rules after conditions in a rule is satisfied.')
+
+    with self.argument_context('network front-door rules-engine rule action', arg_group='Forward Route Override') as c:
+        c.argument('backend_pool', help='Name or ID of a backend pool.', validator=validate_backend_pool)
+        c.argument('forwarding_protocol', arg_type=get_enum_type(FrontDoorForwardingProtocol), help='Protocol to use for forwarding traffic.')
+        c.argument('custom_forwarding_path', help='Custom path used to rewrite resource paths matched by this rule. Leave empty to use incoming path.')
+        c.argument('caching', arg_type=get_three_state_flag(positive_label='Enabled', negative_label='Disabled', return_label=False), help='Whether to enable caching for this route.')
+        c.argument('cache_duration', help='The duration for which the content needs to be cached. Allowed format is ISO 8601 duration')
+        c.argument('dynamic_compression', arg_type=get_three_state_flag(positive_label='Enabled', negative_label='Disabled', return_label=True), help='Use dynamic compression for cached content.')
+        c.argument('query_parameter_strip_directive', arg_type=get_enum_type(FrontDoorQuery), help='Treatment of URL query terms when forming the cache key.')
+        c.argument('query_parameters', help='Query parameters to include or exclude (comma separated).')
```

Comment:
a bit confused for me. How to include and how to exclue?

Reply:
The list of values for example ['key1=value1', 'key2=value2'] will be included or excluded depending on the [query_parameter_strip_directive ](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/frontdoor/resource-manager/Microsoft.Network/stable/2020-01-01/frontdoor.json#L1569) chosen - "StripAllExcept" or "StripOnly" respectively.

Reply:
could you add this relationship between this two arguments into the help message?

Reply:
Yes, good point, added help text in next commit